src/app/components/pages/Approval.jsx

```diff
@@ -16,6 +17,14 @@ class Approval extends React.Component {
     }
 
     render() {
+        if(process.env.BROWSER && this.props.viewMode === VIEW_MODE_WHISTLE) {
+            try {
+                window.postMessage(WHISTLE_SIGNUP_COMPLETE);
+            } catch(e) {
+                console.error('Cannot window.postMessage when not in whistle app');
```

Comment:
why does this try/catch block exist?

Reply:
Because the implementation of postMessage in react-native is different from the spec for web browsers. Calling window.postMessage in a browser with only one argument results in an Error. Because we cannot guarantee a user will never land on this page from a browser, we handle the exception.

Reply:
you're not handling the error, you're ignoring it. why can't you guarantee a user will never land on the page from an error?

Reply:
Can you rephrase
> why can't you guarantee a user will never land on the page from an error?